go-runtime/compile/build.go

```diff
@@ -209,13 +209,30 @@ func Build(ctx context.Context, moduleDir string, sch *schema.Schema, filesTrans
 		}
 		return filesTransaction.ModifiedFiles(filepath.Join(mainDir, "go.mod"), filepath.Join(moduleDir, "go.sum"))
 	})
+	modulesDir := filepath.Join(buildDir, "go", "modules")
 	wg.Go(func() error {
-		modulesDir := filepath.Join(buildDir, "go", "modules")
 		if err := exec.Command(wgctx, log.Debug, modulesDir, "go", "mod", "tidy").RunBuffered(wgctx); err != nil {
 			return fmt.Errorf("%s: failed to tidy go.mod: %w", modulesDir, err)
 		}
 		return filesTransaction.ModifiedFiles(filepath.Join(modulesDir, "go.mod"), filepath.Join(moduleDir, "go.sum"))
 	})
+	modulesDirFiles, err := os.ReadDir(modulesDir)
+	if err != nil && !os.IsNotExist(err) {
+		// could not read external modules directory
+		_ = wg.Wait()
+		return fmt.Errorf("failed to read directory %s: %w", modulesDir, err)
+	}
+	for _, f := range modulesDirFiles {
+		dir := filepath.Join(moduleDir, f.Name())
+		if _, err := os.Stat(filepath.Join(dir, "go.mod")); err == nil {
+			wg.Go(func() error {
+				if err := exec.Command(wgctx, log.Debug, dir, "go", "mod", "tidy").RunBuffered(wgctx); err != nil {
+					return fmt.Errorf("%s: failed to tidy go.mod: %w", dir, err)
+				}
+				return filesTransaction.ModifiedFiles(filepath.Join(dir, "go.mod"), filepath.Join(moduleDir, "go.sum"))
```

Comment:
What's the purpose of a go.mod file within the individual modules vs. at the top level?

Reply:
🤦‍♂️ oh I thought I was tidying the go.mod file that was there, but its one level higher.
I added it because I was getting errors after importing ftl in the external module and this seemed to fix it... But i've since removed it and done a git clean and no such error is occurring now. So all good, nevermind!